config/dynamic.c

```diff
@@ -16,11 +16,534 @@
+		RTE_LOG(WARNING, GATEKEEPER,
+			"Unexpected condition: unknown client type %d at %s\n",
+			client_addr.sun_family, __func__);
+		goto close_fd;
+	}
+
+	/*
+	 * XXX The request must be received under a specified timeout,
+	 * or the request is aborted. This code needs further test.
+	 */
+	ret = setsockopt(conn_fd, SOL_SOCKET, SO_RCVTIMEO,
+		(const char*)&dy_conf->rcv_time_out, sizeof(struct timeval));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVTIMEO) - (%s)\n",
+			strerror(errno));
+		goto close_fd;
+	}
+
+	rcv_buff_size = MSG_MAX_LEN;
+	ret = setsockopt(conn_fd, SOL_SOCKET,
+		SO_RCVBUF, &rcv_buff_size, sizeof(rcv_buff_size));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVBUF) with size = %d - (%s)\n",
+			rcv_buff_size, strerror(errno));
+		goto close_fd;
+	}
+
+	lua_state = luaL_newstate();
+	if (lua_state == NULL) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: failed to create new Lua state!\n");
+		goto close_fd;
+	}
+
+	/* Set up the Lua state while there is a connection. */
+	ret = setup_dy_lua(lua_state);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to set up the lua state\n");
+		goto close_lua;
+	}
+
+	while (1) {
+		ret = process_single_cmd(conn_fd, lua_state);
+		if (ret != 0)
+			break;
+	}
+
+close_lua:
+	lua_close(lua_state);
+
+close_fd:
+	ret = close(conn_fd);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to close the connection socket - (%s)\n",
+			strerror(errno));
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	uint32_t lcore = rte_lcore_id();
+	struct dynamic_config *dy_conf = arg;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	while (likely(!exiting)) {
+		fd_set fds;
+		struct timeval stv;
+ 
+		FD_ZERO(&fds);
+		FD_SET(dy_conf->sock_fd, &fds);
+ 
+		/*
+		 * 1000 usecs' timeout for the select() function.
+		 * This parameter can prevent the select() function
+		 * from blocking forever. So, the whole program can
+		 * exit when receiving a quitting signal.
+		 */
+		stv.tv_sec = 0;
+		stv.tv_usec = 1000;
+
+		ret = select(dy_conf->sock_fd + 1, &fds, NULL, NULL, &stv);
+		if (ret < 0 && errno != EINTR) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"dyn_cfg: Failed to call the select() function - (%s)\n",
+				strerror(errno));
+			break;
+		} else if (unlikely(exiting)) {
+			break;
+		} else if (likely(ret == 0))
+			continue;
+
+		/*
+		 * The config component accepts only one connection at a time.
+		 */
+		if (FD_ISSET(dy_conf->sock_fd, &fds))
+			handle_client(dy_conf->sock_fd, dy_conf);
+	}
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block at lcore = %u is exiting\n", lcore);
+
+	cleanup_dy(dy_conf);
+
+	return ret;
+}
+
+struct dynamic_config *
+alloc_dy_conf(void)
+{
+	return rte_calloc("dynamic_config",
+		1, sizeof(struct dynamic_config), 0);
+}
```

Comment:
Why are we dynamically allocating a `struct dynamic config` if we're always using just one? Should we just have a static one (such as for the LLS).